cpp/src/arrow/compute/CMakeLists.txt

```diff
@@ -18,6 +18,7 @@
 add_custom_target(arrow-compute-tests)
 
 arrow_install_all_headers("arrow/compute")
+arrow_install_all_headers("arrow/compute/kernels")
```

Comment:
Could you remove this to `cpp/src/arrow/compute/kernels/CMakeLists.txt` to install `arrow/compute/kernels/registry.h` to `${PREFIX}/include/arrow/compute/kernels/registry.h`?
If we use this here, `cpp/src/arrow/compute/registry.h` not `cpp/src/arrow/compute/kernels/registry.h` is installed `${PREFIX}/include/arrow/compute/kernels/registry.h`.

Reply:
oh! I see, I misinterpreted how the CMake function worked. Thanks for the pointer @kou! I wasn't really understanding what was the problem and I was pretty sure it was some obvious mistake :)